src/ChannelServer/World/Inventory.cs

```diff
@@ -194,19 +204,71 @@ public Item GetItem(EquipSlot slot)
 		/// </summary>
 		/// <param name="item"></param>
 		/// <return>Index of the item.</return>
-		public int AddSilent(Item item)
+		public List<object> AddSilent(Item item)
 		{
 			var cat = item.Data.Category;
-
 			lock (_syncLock)
 			{
+				List<object> items_updated = new List<object>();
+
 				if (!_items.ContainsKey(cat))
 					throw new ArgumentException("Unknown item category.");
 
-				_items[cat].Add(item);
-				_itemsWorldIndex[item.WorldId] = item;
+				var subIndex = -1;
+				var expected_amount = item.Amount;
+
+				if (item.Data.MaxStack > 1 && this.HasItem(item.Id))
+				{
+					int item_index = subIndex;
+					while (expected_amount > 0)
+					{
+						long inventory_index = this.GetStackableItemIndex(cat, item.Id, item.Amount);
```

Comment:
Why don't you return an int right away, instead of a long that you have to cast to an int before using it?